File: frontend/src/api/lead.ts
import request from './request'

export interface Lead {
  id?: number
  name: string
  company: string
  phone: string
  email: string
  source: string
  status: string
  remark: string
  createdAt?: string
}

export const getLeads = (params?: any) => request.get('/leads', { params })
export const createLead = (data: Lead) => request.post('/leads', data)
export const convertLead = (id: number) => request.put(`/leads/${id}/convert`)

File: backend/src/main/java/com/crm/entity/Lead.java
package com.crm.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("leads")
public class Lead {
    
    @TableId(type = IdType.AUTO)
    private Long id;
    
    @TableField("customer_id")
    private Long customerId;
    
    @TableField("name")
    private String name;
    
    @TableField("company")
    private String company;
    
    @TableField("phone")
    private String phone;
    
    @TableField("email")
    private String email;
    
    @TableField("source")
    private String source;
    
    @TableField("status")
    private String status;
    
    @TableField("score")
    private Integer score;
    
    @TableField("remark")
    private String remark;
    
    @TableField("created_at")
    private LocalDateTime createdAt;
}
